topcoat-data/topcoat-public
# Cleanup files and add some defaults

* Add .DS_Store to .gitignore
* Remove .DS_Store file from root
* Add a .vscode folder to help with default settings across VSCode users 
  * settings.json (inspired by the settings for AppUI: https://github.com/snyk/app-ui/blob/main/.vscode/settings.json) 
  * extensions.json (inspired by the extensions recommended by AppUI: https://github.com/snyk/app-ui/blob/main/.vscode/settings.json) 

* Add eslint configuration from topcoat-core: https://github.com/snyk/topcoat-core/blob/main/frontend/.eslintrc.js

### Suggested workflow
Before starting working on a component, open it in VSCode and save the file so it is formatted. Commit it back to topcoat-public. After it is merged start work. When your work starts, you are starting with a cleanly formatted file. If you work in topcoat-builder and copy and paste code to VSCode you can save again, and it should all get formatted nicely.

Thoughts welcome.

### To Test
* point `config.yml` to `chore/cleanup-files` branch for modules in the `topcoat-builder-ide` when running locally 
```
modules:
    - git: https://github.com/topcoat-data/topcoat-public.git
      revision: chore/cleanup-files
```

* rebuild the modules
* Check to make sure nothing pops up in the 'topcoat-modules' folder in the the builder ide. 
* check to make sure that nothing averse happens when interaticting with pages that import topcoat-public components, and ensure that the pages render correctly.
